### DISCUSSION TITLE
How to pass parameters to an oracle plugin

### DISCUSSION THREAD

**Discussion by pingbaohua** [eclipse-theia/theia#14061] [Q&A]
I added an oracle debug plugin to theia, I want to pass a parameter url to oracle debug plugin, how can I get it in the plugin

**Comment by msujew** [comment]
Hey @pingbaohua,

the API to communicate to VS Code extensions is generally the same as in VS Code. I.e. you will need to use the `CommandRegistry` to call commands that are registered in the VS Code extension. You could pass a parameter that way.

**Reply by pingbaohua** [reply]
I still have no idea, this is the plugin I installed in the Extensions, I want to pass the name field in the url to the Oracle Developer Tools foe vscode plugin, There should be no CommandRegistry setting associated with the Oracle plugin in theia;

![name](https://github.com/user-attachments/assets/5cd5dfec-1372-4f20-a45d-94a01915bf83)

**Reply by msujew** [reply]
Ok, but does the oracle developer tools extension have any API that allows to pass the parameter? I'm not familiar with the extension.
